src/libraries/Microsoft.Extensions.Options/gen/EmitterBase.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Extensions.Options.Generators
+{
+    internal class EmitterBase
+    {
+        private const int DefaultStringBuilderCapacity = 1024;
+        private const int IndentChars = 4;
+
+        private readonly StringBuilder _sb = new(DefaultStringBuilderCapacity);
+        private readonly string[] _padding = new string[16];
+        private int _indent;
+
+        public EmitterBase(bool emitPreamble = true)
+        {
+            var padding = _padding;
+            for (int i = 0; i < padding.Length; i++)
+            {
+                padding[i] = new string(' ', i * IndentChars);
```

Comment:
Can the padding array be anything other than `new string(' ', i * IndentChars)`?

Reply:
I prefer to keep the way is written here. The reason is we can use the indent strings inside interpolated strings too and not only when calling `_sb.Append`.